### PR TITLE
test: remove cachy dependency and cache compatiblity tests

### PR DESCRIPTION
**Pull request by martin-kokos**
# Pull Request Check List

Resolves: kind of related to #5720

Followup of https://github.com/python-poetry/poetry/pull/5868 by chadac which I am happy for. This PR removes `cachy` dependency and compatibility tests that were put in place for the transition from old `cachy.CacheManager` to the new `utils.cache.FileCache` and . Cachy is an abandoned package.

- [ ] ~Added **tests** for changed code.~
- [ ] ~Updated **documentation** for changed code.~

### CODE REVIEW

**Comment by neersighted** [comment]
I'm not sure this is something we want; at least not for now. `cachy` is only used at test-time, and I'd rather have the backwards compat tests for a few more releases to ensure we don't break something. We have cache versioning -- the reason for this code is to make sure we don't choke on the on-disk representation which could be spectacular given that we don't have defensive logic protecting everywhere we use the cache.

To remove cachy, I think I'd prefer to see tests that ensure we gracefully handle a 'corrupt' cache.

**Comment by martin-kokos** [comment]
> I'm not sure this is something we want; at least not for now. `cachy` is only used at test-time, and I'd rather have the backwards compat tests for a few more releases to ensure we don't break something. We have cache versioning above the on-disk format -- the reason for this code is to make sure we don't choke on the on-disk representation which could be spectacular given that we don't have defensive logic protecting everywhere we use the cache.
> 
> To remove cachy, I think I'd prefer to see tests that ensure we gracefully handle a 'corrupt' cache.

Alright. I didn't realize how long term the cache actually is. So my understanding now is that a cache generated by older poetry version can be used by current poetry version. I will attempt to look into that graceful handling.

In the meanwhile, would it be acceptable to modify the tests such that cachy test would be skipable? At this time cachy is imported in conftest, so I don't think there is any good way to achieve that.

**Comment by Secrus** [comment]
> Alright. I didn't realize how long term the cache actually is. So my understanding now is that a cache generated by older poetry version can be used by current poetry version. I will attempt to look into that graceful handling.
> 
> In the meanwhile, would it be acceptable to modify the tests such that cachy test would be skipable? At this time cachy is imported in conftest, so I don't think there is any good way to achieve that.

Would you care to explain why is this such a great matter to you?

**Comment by neersighted** [comment]
It's not that the cache is long-lived, is that historically a invalid cache file jams Poetry until the cache is cleared. As such we need to make sure to feed our replacement for Cachy garbage input and make sure that it gracefully handles corrupt files.

**Comment by martin-kokos** [comment]
> > Alright. I didn't realize how long term the cache actually is. So my understanding now is that a cache generated by older poetry version can be used by current poetry version. I will attempt to look into that graceful handling.
> > In the meanwhile, would it be acceptable to modify the tests such that cachy test would be skipable? At this time cachy is imported in conftest, so I don't think there is any good way to achieve that.
> 
> Would you care to explain why is this such a great matter to you?

I'd like to include poetry in the Gentoo Linux package repo https://github.com/gentoo/gentoo/pull/28843
And because it is desirable to have packages pass their tests in the environment they are installed in, test dependencies need to be satisfied. That would include cachy, but it being an abandoned package and not very popular, I'd like to avoid pulling it into the package repo. It would be acceptable to skip the test involving cachy, but as it is now, I don't think it is possible in a nice way it being imported in conftest. A last resort would be file patching at install time, but that's best to be avoided, and I'd much rather invest the effort in poetry towards as #5720 says *rock solid dependencies*.

**Comment by neersighted** [comment]
Unfortunately, I don't think making our test suite less robust to mitigate distro packaging choices is tenable here in the upstream. If you need to bypass that test, patching it out is your best bet. Or if you want to do something mergable, I'd suggest a robust set of tests with a corpus of good and bad inputs (and maybe even some fuzzing) to ensure that we no longer hang on an invalid cache would be good.

The cachy tests don't prevent the invalid cache issue today, but they do help us avoid making it worse. If we are resilient to an outdated/corrupt on-disk format, that will give us the confidence to drop this set of tests from the test suite.

**Comment by martin-kokos** [comment]
> 

I wouldn't mean to making the tests lest robust in favor of distro packaging. That's why I said, Id rather put the effort into poetry, however unless I am missing something, the new `utils.cache.FileCache` is already being used now https://github.com/python-poetry/poetry/pull/5868 and the tests that this PR removes merely a cleanup of tests for the transition, proving the interfaces of the two cache implementations is the same.

I can understand the need for a more robust cache, but that concern would belong to #5868 (never mind the fact cachy wasn't in the first place, as you've mentioned) or filed as a separate issue.

I've created https://github.com/python-poetry/poetry/pull/7453 in an attempt to make Cache more resilient to cache file corruption.

**Comment by martin-kokos** [comment]
Rebased and resolved conflicts.

Is there any update on the opinion of not wanting to remove tests dependent on cachy just yet?
If not, would it be acceptable to make cachy on *optional* dependency achieved with something like [pytest.importskip](https://docs.pytest.org/en/latest/reference/reference.html#pytest-importorskip)

File: poetry.lock [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.4.0 and should not be changed by hand.
```
Comment (Secrus) [inline poetry.lock:1]: Please revert any changes not connected with the removal of `cachy`.

Comment (martin-kokos) [reply poetry.lock:1]: Since removing it from pyproject.toml, the lock file needs to be regenerated, no?

Comment (Secrus) [reply poetry.lock:1]: yes, but as you can see, you not only removed cachy, but also updated other dependencies, which shouldn't be the case (not as a part of this PR at least)

Comment (martin-kokos) [reply poetry.lock]: Ahh. Did `poetry lock --no-update` and fixed. Thanks.

**Comment by martin-kokos** [comment]
Rebased and resolved conflicts.

**Comment by Secrus** [comment]
@martin-kokos could you rebase and resolve conflicts? I guess it's time to merge this

**Comment by martin-kokos** [comment]
:raised_hands: Rebased. Thanks for the ping.
If you could, please, check a newly added commit resolving some typing checks.

**Review by Secrus — APPROVED** [review]

**Pull request merged**